src/ingest_validation_tools/upload.py

```diff
@@ -422,6 +416,88 @@ def api_validation(
     #
     ##############################
 
+    def _print_multi_assay_data(self):
+        """
+        Print only dataset types of multi_assay_data_paths
+        """
+        print_dict = {}
+        for path, multi_components in self.multi_assay_data_paths.items():
+            print_multi_components = {}
+            for key, value in multi_components.items():
+                if key == "parent":
+                    print_multi_components["parent"] = value[0].dataset_type
+                elif key == "components":
+                    print_components = []
+                    for component in value:
+                        print_components.append(component.dataset_type)
+                    print_multi_components["components"] = print_components
+            print_dict[path] = print_multi_components
+        return print_dict
+
+    def _check_multi_assay_children(
+        self, parent: SchemaVersion, components: List[SchemaVersion]
+    ):
+        """
+        Iterate through child dataset types, check that they are valid
+        components of parent multi-assay type and that no components are missing
+        """
+        not_allowed = []
+        necessary = parent.contains
+        for sv in components:
+            if sv.dataset_type.lower() not in parent.contains:
```

Comment:
I think this flow is why we're limited to one multi-assay dataset type per upload, right? Does this also make it such that we can't have a multi-assay dataset and an unrelated single-assay dataset that is **not** of the same dataset type as that multi-assay dataset's components?

Reply:
Ah that's an oversight, you're right. I will see if I can expand the acceptable cases to accommodate non-parent/non-component inclusions in an upload including a multi-assay.